src/key.cpp

```diff
@@ -172,6 +174,79 @@ bool CKey::Negate()
     return secp256k1_ec_seckey_negate(secp256k1_context_sign, keydata.data());
 }
 
+CKey CKey::AddTweak(const unsigned char *tweak32) const
+{
+    assert(fValid);
+
+    unsigned char tweaked_seckey[32];
+    memcpy(tweaked_seckey, data(), 32);
```

Comment:
In 56882622faf469b6f948f79a69c3c8ddbde92ff8 "crypto: update CKey, CPubkey for silent payments"

We shouldn't be copying secret data out of a secure structure into an insecure one. When `tweaked_seckey` is destroyed, the underlying data may still linger in memory and end up being read by something else.

If this function must return a new `CKey`, then it should instantiate it here and perform the operation on it's `data()` rather than doing this copy.

However, I would prefer if this function mirrored the libsecp one where the tweaking occurs on and modified this `CKey` itself.

Reply:
Now operates on `keydata.data()` in place